swap_meet/vendor.py

```diff
@@ -1,2 +1,82 @@
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        item_list = []
+        for item in self.inventory:
+            if item.category == category:
+                item_list.append(item)
+        return item_list
+
+    def swap_items(self, vendor, my_item, their_item):
+        if their_item not in vendor.inventory or my_item not in self.inventory:
+            return False
+        else:
```

Comment:
Notice here that since this first branch ends in a return, it's essentially a guard clause. This allows us to avoid using the `else` and unindent the block that had been under the else, saving a level of indentation and making it clear what the main logic of the method is.

And unlike the case above where we could use remove directly without checking for the presence of the item first, here, doing the explicit check like this is probably the better option. We want to avoid the situation where we remove _one_ of the items from the first inventory, but then the second removal fails. We _could_ be very careful and put things back they way they were with an error handler, but the preemptive check is more clear.